### DISCUSSION TITLE
How to stream byte[] audio data in Unity app for UWP

### DISCUSSION THREAD

**Discussion by CiEud** [naudio/NAudio#1064] [Q&A]
Hi, I get a byte array and want to play this audio stream in a Unity UWP app (HoloLens 2). When I convert it to floats manually, it is very noisy, so I wanted to use NAudio.
I'm a little confused if I can use NAudio for this and exactly what elements I need for it. Does anyone have a quick reference to the tutorial I can follow? 
I don't have any experience with audio streaming yet. 
The goal is to have audio communication. So I would need to be able to send audio from UWP in byte[] format as well.
Thanks a lot!

**Comment by Axwabo** [comment]
You could try using the [OnAudioFilterRead hook](https://docs.unity3d.com/6000.3/Documentation/ScriptReference/MonoBehaviour.OnAudioFilterRead.html) in which you call an `ISampleProvider`'s `Read` method.
If you meant real-time voice communication, you might be better off writing a custom solution. I recommend using `libopus` and `libcaress` (the latter one for noise suppression)
